lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -174,15 +176,21 @@ def test_symlinks_true(self, stage):
             fs.copy_tree('source', 'dest', symlinks=True)
 
             assert os.path.exists('dest/2')
-            assert os.path.islink('dest/2')
+            if sys.platform != "win32":
+                # TODO: islink will return false for junctions
+                assert os.path.islink('dest/2')
```

Comment:
Should we create a function in `llnl.util.filesystem` like `is_link` which works correctly on Windows and Linux? I think it's worth testing the analogous action on Windows if possible.

Reply:
We do in `llnl.util.symlink`. I don't know why this wasn't changed once that was implemented. Further, refactored, updated, and extended symlink support is included in with the compiler wrapper PR.